bindings/experimental/distrdf/python/DistRDF/__init__.py

```diff
@@ -47,6 +49,113 @@ def initialize(fun, *args, **kwargs):
     from DistRDF.Backends import Base
     Base.BaseBackend.register_initialization(fun, *args, **kwargs)
 
+def DeclareCppCode(code_to_declare: str) -> None:
+    """
+    Declare the C++ code that has to be processed on each worker. 
+    Args:
+        codeToDeclare (str): cpp code to be declared on the workers
+    """
+    from DistRDF.Backends import Base
+    Base.BaseBackend.register_declaration(code_to_declare)
+
+    
+def DistributeHeaders(paths_to_headers: Iterable[str], df = None):
+    """
+    This function allows users to directly load c++ custom headers 
+    onto the workers. 
+
+    Args:
+        paths_to_headers (list): list of paths to headers to be distributed
+        
+        df: dataframe for which the libraries are distributed
+
+    """    
+    from DistRDF.Backends import Base, Utils
+    
+    if df is None:
+        Base.BaseBackend.register_headers(paths_to_headers) 
+
+    else:
+        headers_to_distribute = set()
+        
+        if isinstance(paths_to_headers, str):
+            headers_to_distribute = (Utils.get_paths_set_from_string(paths_to_headers))
+        else: 
+            for path_to_header in paths_to_headers:
+                sanatized_path_to_header = Utils.get_paths_set_from_string(path_to_header)
+                headers_to_distribute.update(sanatized_path_to_header)
+        
+        Utils.declare_headers(headers_to_distribute)
+        df._headnode.backend.distribute_unique_paths(headers_to_distribute)
+        Base.BaseBackend.headers.update(headers_to_distribute)
+
+
+def DistributeFiles(paths_to_files: Iterable[str], df = None):
+    """
+    This function allows users to directly load arbitrary files
+    onto the workers. 
+
+    Args:
+        paths_to_files (list): list of paths to files to be distributed
+        
+        df: dataframe for which the libraries are distributed
+
+    """
+    from DistRDF.Backends import Base, Utils
+    
+    if df is None:
+        Base.BaseBackend.register_files(paths_to_files)
+    
+    else: 
+        files_to_distribute = set()
+        if isinstance(paths_to_files, str):
+            files_to_distribute = (Utils.get_paths_set_from_string(paths_to_files))
+        else: 
+            for path_to_file in paths_to_files:
+                sanatized_path_to_file = Utils.get_paths_set_from_string(path_to_file)
+                files_to_distribute.update(sanatized_path_to_file)
+
+        df._headnode.backend.distribute_unique_paths(files_to_distribute)    
+
+    
+def DistributeSharedLib(paths_to_shared_libraries: Iterable[str], df=None) -> None:
```

Comment:
Also for coherency it should be
```suggestion
def DistributeSharedLibs(paths_to_shared_libraries: Iterable[str], df=None) -> None:
```